### ISSUE TITLE
Runs too fast when SDRAM is enabled

### ISSUE THREAD

**Issue by rsn8887** [MiSTer-devel/Genesis_MiSTer#155]
Compared to real hardware, MiSTer Genesis core runs too fast when SDRAM is enabled.

Speed is correct when DDR Ram is used, as demonstrated here: https://www.youtube.com/watch?v=ebmF_xlRPAA

Tested with Flashback.

**Comment by rsn8887** [comment]
https://twitter.com/SmokeMonsterTWI/status/1341659180717903872?s=19

**Comment by sorgelig** [comment]
it's also related to system bus timings implementation as well, so it will be fixed when system bus will be fixed.

**Comment by ArtemioUrbina** [comment]
Just to add the tests I made to confirm on an independent setup, chain was: Genesis RGB->OSSC->Datapath ES1 and MiSTER analog out via I board -> OSSC -> Datapath ES1. Captures were made with OBS at 60 fps in both

Genesis Model 1 was a VA3, Model 2 is a VA 1.8

Here is the 240p Test Suite running the frame counter on a Genesis vs Mister (baseline, follows properly)
https://www.youtube.com/watch?v=JTh8P_8EQWI&feature=youtu.be

Flash back on a Genesis vs Mister (same as done in OP report but in my setup as confirmation)
https://www.youtube.com/watch?v=J0CFirum91M&feature=youtu.be

Genesis 1 vs Genesis 2 (to discard the YM2612 vs YM3438)
https://www.youtube.com/watch?v=2WlsfwSQf4M&feature=youtu.be

And the Genesis 1 vs MiSTer with SDRAM disabled (RAM DDR3 [Auto disabled])
https://www.youtube.com/watch?v=ebmF_xlRPAA&feature=youtu.be

**Comment by birdybro** [comment]
I was already doing some testing with @donluca today comparing sonic the hedgehog 2, emerald hill zone playback differences between my MD1 VA6 in Sound Test, his MD1 VA1 in Sound Test, and his MD1 VA1 using Deadfish VGM Player with a recent VGM log that is likely accurate. I decided to add recordings from MiSTer DDR3 and SDRAM respectively for fun, just to see how it measures up. The SDRAM in terms of audio is still very accurate (with very slight deviation). With DDR3 it is significantly slower. So if slowing down things to fix this observed behavior with flashback would inadvertently affect audio playback, that should be observed and avoided. If anything audio needs to be just slightly sped up if Sonic 2 is any indication. Attached are audacity project files with the comparison being made.

https://cdn.discordapp.com/attachments/773633550783610973/795743761333354536/bb-ehz-new_and_mister.zip

![image](https://user-images.githubusercontent.com/16388068/103574480-4599e180-4e8d-11eb-9267-cbc7480344de.png)

**Comment by donluca** [comment]
Just to elaborate a little bit: the VGM has been logged by blast'em which has been tested against Valleybell's vgmplayer by converting the VGM to WAV (only for timing purposes, of course): the result in that regard is that the VGM converted to WAV matches exactly, almost sample per sample, the same track ripped from the game's own sound test.
Thus, at least in this context, it proves that blast'em is able to correctly log VGMs with correct timing, unlike KEGA or MAME. This makes blast'em the only emulator so far which is capable of logging "perfect" VGMs.

When moving onto Deadfish VGM Player on the Mega Drive (which has been re-calibrated with MDFourier) the timing was slightly off, but enough to be noticeable by looking at the waveform after a minute or so, which forced me to re-calibrate the DAC scaling on Deadfish again using the aforementioned track.
This has led to a discrepancy against MDFourier of precisely 0.01ms per sample.
I still need to do further investigation on that matter, but take that Deadfish VGM rip with a big pinch of salt.

**Comment by birdybro** [comment]
Yup, I only included the original hardware in the screenshot to point out the difference with DDR3 (and o.g hardware to sdram comparison as a rough control).

**Comment by joshuaryoung** [comment]
Has this been addressed or is this still the case?

**Comment by birdybro** [comment]
Since this time it's a bit more complicated. Deadfish VGM player vs the sound test on Sonic 2 vs In-game music all played at different rates, measurably. Additionally, different hardware does have a slight bit of drift for a few reasons (xtal were not perfectly consistent the way fpga pll calculation is, and xtal do fade over time and slightly modify their speed, all these systems are old, etc...). But the speed difference now with SDRAM is practically indistinguishable and I would consider it within normal hardware differences. PAL is also now correct as tested by a user after I added the PAL mclk by just copying how it was done with the SMS core (they both share the same mclk differences).
